packager/media/formats/mp4/low_latency_segment_segmenter.cc

```diff
@@ -0,0 +1,235 @@
+    file_name_ = options().output_file_name.c_str();
+  } else {
+    file_name_ = GetSegmentName(options().segment_template,
+                               sidx()->earliest_presentation_time,
+                               num_segments_, options().bandwidth);
+  }
+
+  // Create the segment file
+  segment_file_ = File::Open(file_name_.c_str(), "a");
+  if (!segment_file_) {
+    return Status(error::FILE_FAILURE, "Cannot open segment file: " +
+                                            file_name_);
+  }
+
+  std::unique_ptr<BufferWriter> buffer(new BufferWriter());
+  std::unique_ptr<File, FileCloser> file;
+
+  // Point to the already open segment file for writing
+  file.reset(segment_file_);
+  if (!file) {
+    return Status(error::FILE_FAILURE,
+                  "Cannot access the file to write " + file_name_);
+  }
+
+  // Write the styp header to the beginning of the segment.
+  styp_->Write(buffer.get());
+
+  const size_t segment_header_size = buffer->Size();
+  const size_t segment_size = segment_header_size + fragment_buffer()->Size();
+  DCHECK_NE(segment_size, 0u);
+
+  RETURN_IF_ERROR(buffer->WriteToFile(file.get()));
+  if (muxer_listener()) {
+    for (const KeyFrameInfo& key_frame_info : key_frame_infos()) {
+      muxer_listener()->OnKeyFrame(
+          key_frame_info.timestamp,
+          segment_header_size + key_frame_info.start_byte_offset,
+          key_frame_info.size);
+    }
+  }
+
+  // Write the chunk data to the file
+  RETURN_IF_ERROR(fragment_buffer()->WriteToFile(file.get()));
+
+  // Release the file to be used by the next chunk
+  file.release();
+
+  uint64_t segment_duration = GetSegmentDuration();
+  UpdateProgress(segment_duration);
+
+  if (muxer_listener()) {
+    if (!ll_dash_mpd_values_initialized_) {
+      // Set necessary values for LL-DASH mpd after the first chunk has been processed.
+      muxer_listener()->OnSampleDurationReady(sample_duration());
+      muxer_listener()->OnAvailabilityOffsetReady();
+      muxer_listener()->OnSegmentDurationReady();
+      ll_dash_mpd_values_initialized_ = true;
+    }
+    // Add the current segment in the manifest. 
+    // Following chunks will be appended to the open segment file.
+    muxer_listener()->OnNewSegment(file_name_,
+                                  sidx()->earliest_presentation_time,
+                                  segment_duration, segment_size);
+    is_initial_chunk_in_seg_ = false;
+  }
+
+  return Status::OK;
+}
+
+Status LowLatencySegmentSegmenter::WriteChunk() {
+  DCHECK(sidx());
+  DCHECK(fragment_buffer());
+
+  std::unique_ptr<File, FileCloser> file;
+
+  // point to the already open segment file for writing
+  file.reset(segment_file_);
+  if (!file) {
+    return Status(error::FILE_FAILURE,
+                  "Cannot access the file to write " + file_name_);
+  }
+
+  // Write the chunk data to the file
+  RETURN_IF_ERROR(fragment_buffer()->WriteToFile(file.get()));
```

Comment:
Where/how does fragment_buffer get reset?

Reply:
The `fragment_buffer_` is type `BufferWriter`.  The `BufferWriter` method `WriteToFile(File* file)`  clears the buffer after writing the buffer's contents to `file`. See [line 85 of buffer_writer.cc](https://github.com/google/shaka-packager/blob/f018c9a9bf6d936f3434ea32fb6a9ee42ae12b4b/packager/media/base/buffer_writer.cc#L85).

Reply:
Perfect, thanks!  I don't know all the things.  :-)

Reply:
No worries! I had to hunt down the buffer clearing. A lot of stuff is quite hidden - like the `delete` in `File::Close( )` 😅.